samples/dnn/scene_text_detection.cpp

```diff
@@ -0,0 +1,380 @@
+        "{ unclipRatio ratio                |2.0| unclip ratio. }"
+        "{ evaluate e                       |false| false: predict with input images; true: evaluate on benchmarks. }"
+        "{ evalDataPath dp                  | | Path to benchmarks for evaluation. "
+            "Download link: https://drive.google.com/file/d/149tAhIcvfCYeyufRoZ9tmc2mZDKE_XrF/view?usp=sharing}";
+
+
+void getTextPolygons(const Mat & binary, float binThresh, float polyThresh, size_t maxCandidates,
+                     std::vector<std::vector<Point>> & ploygons, int outputType, double unclipRatio);
+
+double boxFastScore(const Mat & binary, std::vector<Point> & contour);
+
+void unclip(std::vector<Point> &inPoly, std::vector<Point> &outPoly, double ratio);
+
+int main(int argc, char** argv)
+{
+    // Parse arguments
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("Use this script to run the official PyTorch implementation (https://github.com/MhLiao/DB) of "
+                 "Real-time Scene Text Detection with Differentiable Binarization (https://arxiv.org/abs/1911.08947)\n"
+                 "The current version of this script is a variant of the original network w/o deformable convolution");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    float binThresh = parser.get<float>("binaryThreshold");
+    float polyThresh = parser.get<float>("polygonThreshold");
+    uint maxCandidates = parser.get<uint>("maxCandidate");
+    String modelPath = parser.get<String>("modelPath");
+    int outputType = parser.get<int>("outputType");
+    double unclipRatio = parser.get<double>("unclipRatio");
+    int height = parser.get<int>("inputHeight");
+    int width = parser.get<int>("inputWidth");
+
+    if (!parser.check())
+    {
+        parser.printErrors();
+        return 1;
+    }
+
+    // Load the network
+    CV_Assert(!modelPath.empty());
+    Net detector = readNet(modelPath);
+
+    // Create a window
+    static const std::string winName = "Real-time Scene Text Detection with Differentiable Binarization";
+    namedWindow(winName, WINDOW_NORMAL);
+
+    if (parser.get<bool>("evaluate")) {
+        // for evaluation
+        String evalDataPath = parser.get<String>("evalDataPath");
+        CV_Assert(!evalDataPath.empty());
+        String testListPath = evalDataPath + "/test_list.txt";
+        std::ifstream testList;
+        testList.open(testListPath);
+        CV_Assert(testList.is_open());
+
+        // Create a window for show groundtruth
+        static const std::string winNameGT = "GT";
+        namedWindow(winNameGT, WINDOW_NORMAL);
+
+        String testImgPath;
+        while (std::getline(testList, testImgPath)) {
+            String imgPath = evalDataPath + "/test_images/" + testImgPath;
+            // std::cout << "Image Path: " << imgPath << std::endl;
+
+            Mat frame = imread(imgPath, 1);
+            CV_Assert(!frame.empty());
```

Comment:
You could use something like https://github.com/opencv/opencv/blob/73959fed4581556d8e6b770987c819adfacceb87/modules/video/perf/perf_bgfg_mog2.cpp#L30

Reply:
I feel confused that this file is only a sample for scene text detection, not an API. Is there anything related to the test?
Why should I change CV_Assert into SkipTestException?
Thank you.

Reply:
Yes check only for test.